src/test/java/org/opentripplanner/middleware/controllers/api/ManageLegTraversalTest.java

```diff
@@ -0,0 +1,197 @@
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opentripplanner.middleware.models.MonitoredTrip;
+import org.opentripplanner.middleware.models.TrackedJourney;
+import org.opentripplanner.middleware.testutils.CommonTestUtils;
+import org.opentripplanner.middleware.triptracker.ManageLegTraversal;
+import org.opentripplanner.middleware.triptracker.ManageTripTracking;
+import org.opentripplanner.middleware.triptracker.TrackingLocation;
+import org.opentripplanner.middleware.triptracker.TripStatus;
+import org.opentripplanner.middleware.utils.Coordinates;
+import org.opentripplanner.middleware.utils.DateTimeUtils;
+import org.opentripplanner.middleware.utils.JsonUtils;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.opentripplanner.middleware.triptracker.ManageLegTraversal.getSecondsToMilliseconds;
+import static org.opentripplanner.middleware.triptracker.ManageLegTraversal.interpolatePoints;
+
+public class ManageLegTraversalTest {
+
+    private static MonitoredTrip monitoredTrip;
+    private static MonitoredTrip busStopJusticeCenterTrip;
+
+    private static final int SUBWAY_LEG_ID = 1;
+
+    private static final int FINAL_LEG_ID = 2;
+
+    @BeforeAll
+    public static void setUp() throws IOException {
+        monitoredTrip = JsonUtils.getPOJOFromJSON(
+            CommonTestUtils.getTestResourceAsString("controllers/api/monitored-trip-for-tracking.json"),
+            MonitoredTrip.class
+        );
+        busStopJusticeCenterTrip = JsonUtils.getPOJOFromJSON(
+            CommonTestUtils.getTestResourceAsString("controllers/api/bus-stop-justice-center-trip.json"),
+            MonitoredTrip.class
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("createTrace")
+    void canTrackTrip(String time, double lat, double lon, TripStatus expected) {
+        TrackedJourney trackedJourney = new TrackedJourney();
+        trackedJourney.tripId = busStopJusticeCenterTrip.id;
+        TrackingLocation trackingLocation = new TrackingLocation(time, lat, lon);
+        trackedJourney.locations = List.of(trackingLocation);
+        TripStatus tripStatus = ManageTripTracking.getTripStatus(trackedJourney, busStopJusticeCenterTrip);
+        assertEquals(expected, tripStatus);
+    }
+
+    private static Stream<Arguments> createTrace() {
+        return Stream.of(
+            Arguments.of("2024-01-26T19:07:05Z", 33.95022, -83.9906, TripStatus.BEHIND_SCHEDULE),
+            Arguments.of("2024-01-26T19:07:08Z", 33.95022, -83.9906, TripStatus.ON_SCHEDULE),
+            Arguments.of("2024-01-26T19:07:50Z", 33.95022, -83.9906, TripStatus.AHEAD_OF_SCHEDULE),
+            Arguments.of("2024-01-26T19:07:43Z", 33.6408211227185, -84.4465224660685, TripStatus.DEVIATED),
```

Comment:
Maybe insert a comment, especially this case, for where the coordinate is with respect to the leg geometry.